cpp/include/cudf/types.h

```diff
@@ -72,27 +72,50 @@ typedef enum {
     GDF_HASH_IDENTITY,  /**< Identity hash function that simply returns the key to be hashed */
 } gdf_hash_func;
 
+
+/**
+ * @brief Defines the unit of time that an algoithm or structure is storing.
+ *
+ * There are time types in cudf. Those time types can have different resolutions so this
+ * enum allows us to define different units of time. The types included are nanosecond,
+ * microsecond, millisecond, and second.
+ */
 typedef enum {
-	TIME_UNIT_NONE=0, // default (undefined)
-	TIME_UNIT_s,   // second
-	TIME_UNIT_ms,  // millisecond
-	TIME_UNIT_us,  // microsecond
-	TIME_UNIT_ns   // nanosecond
+	TIME_UNIT_NONE=0, /**< The default time unit type. */
+	TIME_UNIT_s,   /**< Second resolution time unit type */
+	TIME_UNIT_ms,  /**< Millisecond resolution time unit type */
+	TIME_UNIT_us,  /**< Microsecond resolution time unit type */
+	TIME_UNIT_ns   /**< Nanosecond resolution time unit type */
 } gdf_time_unit;
 
+/**
+ * @brief This struct allows us to include extra information about a columns type.
+ *
+ * Column types sometimes need extra information to properly encode the representation
```

Comment:
These three lines add no useful information, are verbose, and use the "royal us". Delete them and document the meaning of `time_unit`, preferably with an inline `///<`.